src/NServiceBus.Metrics/MetricsFeature.cs

```diff
@@ -1,63 +1,102 @@
-﻿using System.Threading.Tasks;
+﻿using System.Linq;
+using System.Threading.Tasks;
 using Metrics;
 using NServiceBus;
 using NServiceBus.Features;
-using NServiceBus.Metrics;
+using NServiceBus.Hosting;
 using NServiceBus.Metrics.QueueLength;
 using NServiceBus.ObjectBuilder;
+using NServiceBus.Transport;
 
 class MetricsFeature : Feature
 {
-    public MetricsFeature()
-    {
-        Defaults(s => { s.Set<MetricsRegistry>(new MetricsRegistry()); });
-    }
-
     protected override void Setup(FeatureConfigurationContext context)
     {
         context.ThrowIfSendonly();
-
-        var resetMetricTimer = new ResetMetricTimer(context);
-
-        context.RegisterMetricBuilder(new CriticalTimeMetricBuilder(resetMetricTimer));
-        context.RegisterMetricBuilder(new PerformanceStatisticsMetricBuilder());
-        context.RegisterMetricBuilder(new ProcessingTimeMetricBuilder(resetMetricTimer));
-        context.RegisterMetricBuilder(new QueueLengthMetricBuilder());
+        
+        var probeContext = BuildProbes(context);
 
         var settings = context.Settings;
         var metricsOptions = settings.Get<MetricsOptions>();
 
+        metricsOptions.SetUpObservers(probeContext);
+
         // the context is used as originating endpoint in the headers
         MetricsContext metricsContext = new DefaultMetricsContext($"{settings.EndpointName()}");
 
-        ConfigureMetrics(context, metricsContext);
+        SetUpQueueLengthReporting(context, metricsContext);
+
+        SetUpSignalReporting(probeContext, metricsContext);
 
-        context.RegisterStartupTask(builder => new MetricsReporting(metricsContext, metricsOptions, builder));
+        if (!string.IsNullOrEmpty(metricsOptions.ServiceControlMetricsAddress))
```

Comment:
It's possible to use the metrics without a metrics server, metrics server is only for sending the data to e.g. SP.